### ISSUE TITLE
How do I change the style for the bounding box and mask?

### ISSUE THREAD

**Issue by schosdas** [ultralytics/ultralytics#9146]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,
I attempted to utilize the Yolov8-seg model and OpenCV to exhibit inference results on a video. In this context, I aim to introduce transparency to the mask section of the segmentation and eliminate certain text output. Could you kindly instruct me on the necessary code modifications to address this for output styles? 
My code is provided below.

```
import cv2
from ultralytics import YOLO
import torch
import numpy as np
from collections import defaultdict 

# custom model load
model = YOLO("model/yolov8_model.pt")

cap = cv2.VideoCapture("test_video7.avi")

track_history = defaultdict(lambda: []) 

while cap.isOpened():
    ret, frame = cap.read()
    if not ret:
        break

    results = model.track(frame, persist=False) 

    boxes = results[0].boxes.xywh.cpu()
    track_ids = results[0].boxes.id.int().cpu().tolist()

    annotated_frame = results[0].plot()

    #  (Use X)
    # for box, track_id in zip(boxes, track_ids):
    #     x, y, w, h = box 
    #     track = track_history[track_id] 
    #     track.append((float(x), float(y)))  
    #
    #     if len(track) > 30:
    #         track.pop(0)
    #
    #     # Draw the tracking lines 
    #     # points = np.hstack(track).astype(np.int32).reshape((-1, 1, 2))
    #     # cv2.polylines(annotated_frame, [points], isClosed=False, color=(230, 230, 230), thickness=10)

    # Display the annotated frame
    cv2.imshow("YOLOv8 Tracking", annotated_frame)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()
```

**Comment by glenn-jocher** [comment]
@schosdas hey there! 🌟

For customizing the bounding box and mask styles, including transparency for the mask and modifying text outputs, you'll need to adjust the `plot()` method in your code. Unfortunately, this can't be directly configured through the API or your current script. 

However, for transparency, a common approach involves manual overlaying. You can adjust the `plot()` method source code to set transparency levels (alpha value) for masks. For textual output, similarly, you would have to modify or conditionally remove the text drawing commands in the source.

While I can't offer a snippet to directly modify the `plot()` as it's part of the YOLO package internals, the general idea for transparency would look something like this with OpenCV:

```python
alpha = 0.5  # Transparency factor.
# Assuming 'mask' is your mask array and 'frame' is the frame you want to overlay it on
frame_overlay = frame.copy()
cv2.addWeighted(mask, alpha, frame_overlay, 1 - alpha, 0, frame_overlay)
```

For removing text, you would essentially skip or comment out the commands that draw text in the `plot()` method's implementation.

These changes imply modifying the library code itself, so proceed with caution and maybe back up the original files. Alternatively, after `results` computation, you might overlay custom masks and annotations manually, leveraging OpenCV directly for full control.

Feel free to explore the YOLO source or ask the community for more targeted customizations. Happy coding! 🚀

**Comment by mudassir659** [comment]
how do i change the prediction box line width in yolov8

**Comment by glenn-jocher** [comment]
@mudassir659 hi there! 👋

To change the prediction box line width in YOLOv8 when plotting results, you can use the `line_width` parameter in the `plot()` method. Here's a quick example:

```python
results = model.predict('path_to_your_image.jpg')
annotated_frame = results[0].plot(line_width=3)  # Adjust line_width as needed
```

Just replace `3` with your preferred line width. Hope this helps! Let us know if you have more questions. Happy coding! 🚀
